src/XrdTpc/XrdTpcUtils.cc

```diff
@@ -20,14 +20,17 @@
 //------------------------------------------------------------------------------
 #include "XrdTpcUtils.hh"
 #include "XrdOuc/XrdOucTUtils.hh"
+#include "XrdTpc/XrdTpcTPC.hh"
 
 #include <sstream>
 
-std::string XrdTpcUtils::prepareOpenURL(const std::string & reqResource, std::map<std::string,std::string> & reqHeaders, const std::map<std::string,std::string> & hdr2cgimap, bool & hasSetOpaque) {
+std::string XrdTpcUtils::prepareOpenURL(const std::string & reqResource, std::map<std::string,std::string> & reqHeaders, const std::map<std::string,std::string> & hdr2cgimap) {
   auto iter = XrdOucTUtils::caseInsensitiveFind(reqHeaders,"xrd-http-query");
-  bool found_first_header = false;
   std::stringstream opaque;
 
+  // https://github.com/xrootd/xrootd/issues/2427 we tell the oss layer that this transfer is a HTTP TPC one
+  opaque << "?" << TPC::TPCHandler::OSS_TASK_OPAQUE;
```

Comment:
Why not do `opaque << "?oss.task=httptpc";` here and get rid of `TPC::TPCHandler::OSS_TASK_OPAQUE`?

Reply:
HAHA I initially had this duplication everywhere and hesitated to push without putting this variable. I wanted to remove the duplication. As you prefer if you prefer I can duplicate without issue!

Reply:
I don't mind it as is, but duplicating makes the code more readable. If you don't want to revert the work for deduplicating, we can merge as is. Just waiting on ack from @abh3.

Reply:
Hmm, let's see a) use a declared static, b) use define symbol with a shorter name, or c) use a string constant. What's wrong with each? a) not particularly clear and rather clunky, b) people hate define symbols because of namespace issues, c) easy to make a mistake because of duplication. So, it's a wash; any approach is just as good or bad, depending on how you look at it  So, I'm neutral. Frankly, I'd just keep it as it is.